### ISSUE TITLE
feat(v2): add fallback to BrowserOnly component

### ISSUE THREAD

**Issue by nebrelbug** [facebook/docusaurus#2573]
## 🚀 Feature

Fallback support for the BrowserOnly component, for SEO reasons.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

I'm embedding RunKit playgrounds on my Docusaurus v2 sites (https://eta.js.org) and (https://squirrelly.js.org). I want fallback content to render if the user doesn't have JavaScript enabled, etc.

## Pitch

Basically, you could optionally pass a `fallback` prop to the `<BrowserOnly>` component, and the fallback would render if on SSR.

```jsx
import BrowserOnly from '@docusaurus/BrowserOnly';

function MyComponent() {
  return (
    <BrowserOnly fallback={<p>Some Fallback Content</p>}>
      {() => {
        // Something that should be excluded during build process prerendering.
      }}
    </BrowserOnly>
  );
}
```

I'd love to open a PR for this!

**Comment by yangshun** [comment]
Sounds reasonable. I like that this API is similar to `<Suspense fallback={...}>` (as of now).
